content/root/best-practice-guide.textile

```diff
@@ -19,28 +19,57 @@ h2(#ably-101). Ably 101
+Ably Client Library SDKs may implement two interfaces: realtime and REST. The "Realtime":/realtime  interface maintains a persistent connection to Ably and can be considered stateful, whereas the "REST":/rest interface issues HTTP requests to the Ably platform and is stateless. The REST interface is therefore much lighter weight as it does not need the complexity of maintaining state. The REST interface can also only publish, but not subscribe to messages.
+
+Depending on your use case, you may only need to use the REST interface on your server, as it may only need to authenticate with Ably. You typically use the realtime interface on your client.
 
-h3(#auth). Authentication and Encryption
+The PHP and Python SDKs only have a REST interface. While the Ruby SDK implements both realtime and REST interfaces, there is also a Ruby SDK available that implements only the REST interface. 
 
-Ably supports two main means of authentication: "Basic Authentication":/core-features/authentication#basic-authentication and "Token Authentication":/core-features/authentication#token-authentication. Basic Authentication is very easy to set up and requires the API key to be used by your clients. Token Authentication allows you to connect your auth server to Ably, using the API key, which allows your clients to handle authentication via the short lived tokens. These tokens provide a secure way to pass information over the internet and make sure that the actual API key is never exposed to your end clients but remains safely with your auth server. If you prefer to use a single auth strategy across all third party services, you can use "JWT Authentication":/core-features/authentication#ably-jwt-process as part of Token Authentication.
+An important difference between the REST and realtime interface is that with REST you can publish but not subscribe to realtime messages. With the realtime interface you can both publish and subscribe. If you have a use case where you only need to authenticate, or publish messages, then the REST interface is recommended.
+
+h3(#encryption). Encryption
 
 Ably supports "TLS being enabled by default":https://knowledge.ably.com/are-messages-sent-to-and-received-from-ably-securely-using-tls. This ensures that data is securely sent to and received from Ably. However, messages are not encrypted within the Ably system. Using the "encryption feature":/realtime/encryption of our client libraries, you can make sure that message payloads are opaque, can never be decrypted by Ably but only by other clients that share your secret key.
 
+h3(#pubsub). Publish and Subscribe (Pub/Sub)
+
+You can publish data to named "channels":/realtime/channels within Ably's platform and the platform will make sure that all clients subscribed to these channels will receive the data in realtime.
```

Comment:
```suggestion
You can publish data to named "channels":/realtime/channels within Ably's platform and the platform will make sure that all clients subscribed to those channels will receive the data in realtime.
```

Reply:
This feels like it's the ultimate aim of all the other things we discuss in this section and should therefore be at the top, before authentication?